isis/scripts/isisVarInit.py

```diff
@@ -20,12 +22,50 @@
 #
 # SPDX-License-Identifier: CC0-1.0
 
+# This is the full path to this environment:
+CONDA_PREFIX = os.environ["CONDA_PREFIX"]
+ENVS_DIR = Path(CONDA_PREFIX).parent
+THIS_ENV = Path(CONDA_PREFIX).name
+
+
+def get_conda_envs() -> List[Path]:
+    """Returns a list of paths to the found conda envs.
+
+    This method is used because
+    (1) there is no API for `conda env`,
+    (2) `subprocess.check_output` and `sh.conda` are not producing the same output.
+    """
+    print(f"Searching for other conda envs in {ENVS_DIR} ...")
+    paths = [d for d in ENVS_DIR.iterdir() if d.is_dir()]
+    return [p.name for p in paths if p.name != THIS_ENV]
+
+
+def get_envs_to_init() -> List[Path]:
+    """Find conda envs and ask user which ones should initialize ISIS."""
+    env_names = get_conda_envs()
+    print("Found the following other conda envs:")
+    for i, env in enumerate(env_names):
+        print(i, env)
+    print("Which envs should receive the ISIS path initialization?")
+    print("(Press RETURN for skipping.)")
+    answer = ""
+    answer = input("Comma-separated list of env indexes (e.g. 1,2,4):")
+    if not answer:
+        return []
+    try:
+        envs_to_init = [env_names[int(i)] for i in answer.split(",")]
+    except ValueError:
+        raise ValueError(
+            "Could not parse your answer.\nDid you use a comma-separated list?"
+        )
+    return envs_to_init
```

Comment:
If I were to run this, I would have to y/n through 60+ environments. Perhaps it is better to take an optinal list of envs as arguments to the script?

I realize that the counter argument to this is that maybe I do want ISIS in all 60. So impossible to make everyone happy. Maybe the solution is to use either a list or an all flag.

Reply:
it's not per env, you provide a comma-separated list of the printed indexes in the UI dialog.
However, I think taking the actual env names as comma-separated lists would be better and less error prone than using indexes. More work for the user, but it would allow you to paste in a list of envs, yes.